blueflood-core/src/main/java/com/rackspacecloud/blueflood/service/ShardStateManager.java

```diff
@@ -326,12 +322,22 @@ protected UpdateStamp getAndSetState(int slot, UpdateStamp.State state) {
                     //Since we only allow delayed metrics upto 3 days(BEFORE_CURRENT_COLLECTIONTIME_MS), a slot can be
                     //identified as being re-rolled, if the last rollup is within those last 3 days. (theoretically, if
                     //there are no delayed metrics, a slot should only get rolled again after 14 days since its last rollup)
-                    if (timeElapsedSinceLastRollup < DELAYED_METRICS_MAX_ALLOWED_DELAY &&
-                            timeElapsed <= delayedMetricsMaxAgeMillis) {
-                        log.debug(String.format("Delaying rollup for shard:[%s], slotState:(%s,%s,%s) as [%d] millis " +
-                                                "havent elapsed since last rollup:[%d]", shard, granularity, entry.getKey(),
-                                                update.getState().code(), delayedMetricsMaxAgeMillis, update.getLastRollupTimestamp()));
-                        continue;
+                    if (timeElapsedSinceLastRollup < DELAYED_METRICS_MAX_ALLOWED_DELAY) {
+
+                        if (timeElapsed <= delayedMetricsMaxAgeMillis) {
+
+                            SlotKey slotKey = SlotKey.of(granularity, entry.getKey(), shard);
+                            log.debug(String.format("Delaying re-roll of slotKey [%s] as [%d] millis haven't elapsed " +
+                                    "since collection time:[%d] now: [%d], last rollup time: [%d]", slotKey,
+                                    delayedMetricsMaxAgeMillis, update.getTimestamp(), now, update.getLastRollupTimestamp()));
+                            continue;
+                        } else {
+
+                            granToReRollMeters.get(granularity).mark();
+                            if (nowMillis - update.getTimestamp() >= millisInADay) {
```

Comment:
Thats existing logic. I just moved it around as I thought there is a better way of determining re-rolls.